src/components/newcases.js

```diff
@@ -0,0 +1,94 @@
+import React, {useState, useEffect} from 'react';
+
+function RateOfNewCases(props) {
+  const [casesMap, setCasesMap] = useState(0);
+  const upArrow = String.fromCharCode(8593);
+  const downArrow = String.fromCharCode(8595);
+
+  useEffect(() => {
+    parseData();
+  }, [props.timeseries, props.deltas]);
+
+  const parseData = () => {
+    const length = props.timeseries.length;
+    if (length > 0) {
+      const casesDetails = [
+        {
+          cases: props.deltas.confirmeddelta,
+          date: 'Today',
+        },
+        {
+          cases: props.timeseries[length - 1]['dailyconfirmed'],
+          date: 'Yesterday',
+        },
+        {
+          cases: props.timeseries[length - 2]['dailyconfirmed'],
+          date: props.timeseries[length - 2]['date'],
+        },
+        {
+          cases: props.timeseries[length - 3]['dailyconfirmed'],
+          date: props.timeseries[length - 3]['date'],
+        },
+        {
+          cases: props.timeseries[length - 4]['dailyconfirmed'],
+          date: props.timeseries[length - 4]['date'],
+        },
+      ];
+      setCasesMap(casesDetails);
+    }
+  };
+
+  function displayRateOfNewCases() {
+    const rows = [];
+    for (let index = casesMap.length - 1; index > 0; index--) {
+      const casesDiff = casesMap[index - 1].cases - casesMap[index].cases;
+      const isToday = casesMap[index - 1].date === 'Today';
+      rows.push(
+        <React.Fragment key={index}>
+          {index < casesMap.length - 1 && (
+            <div
+              className={`level-item ${
+                casesDiff < 0 ? (isToday ? 'is-blue' : 'is-green') : 'is-cherry'
```

Comment:
This can be moved into variable as well. It's repeated in line 65.

Reply:
done